src/scheduling/appointments/appointment-slice.ts

```diff
@@ -72,12 +78,19 @@ export const fetchAppointment = (id: string): AppThunk => async (dispatch) => {
 export const createAppointment = (
   appointment: Appointment,
   onSuccess?: (appointment: Appointment) => void,
+  onError?: () => void,
 ): AppThunk => async (dispatch) => {
-  dispatch(createAppointmentStart())
-  const newAppointment = await AppointmentRepository.save(appointment)
-  dispatch(createAppointmentSuccess())
-  if (onSuccess) {
-    onSuccess(newAppointment)
+  try {
+    const newAppointment = await AppointmentRepository.save(appointment)
+    dispatch(createAppointmentStart())
+    dispatch(createAppointmentSuccess())
+    if (onSuccess) {
+      onSuccess(newAppointment)
+    }
+  } catch (e) {
+    if (onError) {
+      onError()
```

Comment:
Is it possible to pass the error `e` to the error handler?

Reply:
At the moment, no. Since this function just renders a toast component with an error message